onmt/transforms/tokenize.py

```diff
@@ -305,16 +353,29 @@ def _get_subword_kwargs(self, side='src'):
             kwopts['sp_alpha'] = subword_alpha
         else:
             logger.warning('No subword method will be applied.')
+        vocabulary_threshold = self.tgt_vocab_threshold if side == 'tgt' \
+            else self.src_vocab_threshold
+        vocabulary_path = self.tgt_subword_vocab if side == 'tgt' \
+            else self.src_subword_vocab
+        if vocabulary_threshold > 0 and vocabulary_path != "":
+            kwopts['vocabulary_path'] = vocabulary_path
```

Comment:
So, we have some 'first level' opts that are added in the `add_options` method, but these remain in the `src/tgt_onmttok_kwargs` dict string? Not very user friendly I reckon. Shouldn't these vocabulary_X be 'first level' as well? (If we say all the subword related stuff is there.)

Reply:
the `*_onmttok_kwargs` are intended to pass all non-subword related options (that not specified in the Common part) and since `vocabulary_path` and `vocabulary_threshold` are subword related, it's should not be included in `*_onmttok_kwargs`.
This `_get_subword_kwargs` is intended to gather these subword-related options to each side, so I find by doing this way is rational.
And any attempt to set the `src_subword_vocab` and also includes `vocabulary_path` in the `src_onmttok_kwargs` will raise `SyntaxError`, so should have no risk.